File: server/src/main/java/com/hospitaltender/server/service/SupplierService.java
package com.hospitaltender.server.service;

import com.hospitaltender.server.dto.request.CreateSupplierRequest;
import com.hospitaltender.server.dto.request.UpdateSupplierRequest;
import com.hospitaltender.server.dto.response.SupplierResponse;

import java.util.List;

public interface SupplierService {

    SupplierResponse create(CreateSupplierRequest request);

    SupplierResponse getById(Long id);

    List<SupplierResponse> getAll();

    List<SupplierResponse> getActiveSuppliers();

    SupplierResponse update(Long id, UpdateSupplierRequest request);

    void delete(Long id);
}

File: ui/hospital-system-ui/src/services/supplier.service.ts
import api from './api'
import { SupplierResponse, CreateSupplierRequest, UpdateSupplierRequest } from '@/types'

export const supplierService = {
  getAll: () =>
    api.get<SupplierResponse[]>('/suppliers').then((res) => res.data),

  getById: (id: number) =>
    api.get<SupplierResponse>(`/suppliers/${id}`).then((res) => res.data),

  create: (data: CreateSupplierRequest) =>
    api.post<SupplierResponse>('/suppliers', data).then((res) => res.data),

  update: (id: number, data: UpdateSupplierRequest) =>
    api.put<SupplierResponse>(`/suppliers/${id}`, data).then((res) => res.data),

  delete: (id: number) =>
    api.delete(`/suppliers/${id}`),
}
